docs/drafts/derive-serviceaccount.md

```diff
@@ -0,0 +1,578 @@
+  name: <cluster-extension-name>
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: <cluster-extension-name>
+subjects:
+- kind: ServiceAccount
+  name: <cluster-extension-sa-name>
+  namespace: <cluster-extension-namespace>
+```
+
+Below is the translation of the above cluster permissions into Role and RoleBinding for your cluster extension:
+
+```yml
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
+  name: <cluster-extension-name>
+  namespace: <cluster-extension-name>
+rules:
+- apiGroups:
+  - ""
+  resources:
+  - pods
+  - services
+  - services/finalizers
+  - endpoints
+  - persistentvolumeclaims
+  - events
+  - configmaps
+  - secrets
+  verbs:
+  - create
+  - delete
+  - get
+  - list
+  - patch
+  - update
+  - watch
+- apiGroups:
+  - apps
+  resources:
+  - deployments
+  verbs:
+  - create
+  - delete
+  - get
+  - list
+  - patch
+  - update
+  - watch
+- apiGroups:
+  - apps
+  resourceNames:
+  - <cluster-extension-name>
+  resources:
+  - deployments/finalizers
+  verbs:
+  - update
+- apiGroups:
+  - ""
+  resources:
+  - pods
+  verbs:
+  - get
+- apiGroups:
+  - apps
+  resources:
+  - deployments
+  verbs:
+  - get
+- apiGroups:
+  - ""
+  resources:
+  - serviceaccounts
+  verbs:
+  - create
+  - get
+  - list
+  - patch
+  - update
+  - watch
+- apiGroups:
+  - rbac.authorization.k8s.io
+  resources:
+  - roles
+  - rolebindings
+  verbs:
+  - create
+```
+Below is the RoleBinding that associates your cluster extension to its service account:
+
+```yml
+apiVersion: rbac.authorization.k8s.io/v1
+kind: RoleBinding
+metadata:
+  name: <cluster-extension-name>
+  namespace: <cluster-extension-namespace>
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: Role
+  name: <cluster-extension-name>
+subjects:
+- kind: ServiceAccount
+  name: <cluster-extension-sa-name>
+```
+
+You can use the above example as a starting point to derive the minimal RBAC for your cluster extension.
+You should start by specifying permissions to manage the Deployment, ServiceAccount and Roles required for your cluster extension. Subsequently, you can iterate as below.
+
+* Create all the initial RBAC and then iterate over the ClusterExtension failures, examining conditions and updating the RBAC to include the generated cluster role names (name will be in the failure condition).
+* After reading the failure condition, update the installer RBAC and iterate until you are out of errors.
+* You can get the bundle image, unpack the same and inspect the manifests to determine the required permissions.
+* The `oc` cli-tool creates cluster roles with a hash in their name. You can query the newly created ClusterRole names and reduce the installer RBAC scope to have the ClusterRoles needed, this can include generated roles.
+* You can achieve this by allowing the installer to get, list, watch and update any cluster roles.
+
+Note: Production tools to help you manage RBAC are also available with OLM v1 release.
```

Comment:
Unless something has changed, it's my understanding that we no _not_ yet have production grade tools. This would be a good section to mention the hack/tools/catalogs (non production tools) that can be used to help with the process.